### PR TITLE
Page controller refactor

### PR DESCRIPTION
**Pull request by emodric**
> Repeat of #1457

This PR refactors page controller to use the View API.

It also adds block cache response listener and view parameters injection listeners.

### CODE REVIEW

File: doc/bc/changes-6.0.md [hunk 1]
```diff
@@ -211,6 +211,13 @@ Changes affecting version compatibility with former or future versions.
   Those need to be changed to custom content view controllers, that use a contentId instead of a locationId as an
   argument. The location is available in the `$parameters` array.
 
+* The `viewBlock` and `viewBlockById` `PageController` actions are deprecated.
+  `viewAction` must be used instead. It accepts both `block` and `id` as parameters.
```
Comment (andrerom) [inline doc/bc/changes-6.0.md]: should we mention `block` here? isn't that handled by parameter converter? *as in, wasn't that a bug some years ago, people passing object as param*

Comment (emodric) [inline doc/bc/changes-6.0.md]: `viewAction` and `embedAction` docs mention location object as an extra param too :)

I would mention them, I see no reason why not. If you're thinking what I'm thinking, the bug was actually inability to use `render_esi` with `ez_page:viewBlock` as it only accepted block objects and not block IDs. That was solved by adding `viewBlockById` method in `ez_page` controller.

**Comment by emodric** [comment]
The failures seem to be unrelated to this :)

**Comment by bdunogier** [comment]
This looks good to me, +1.

@lolautruche ?

File: eZ/Bundle/EzPublishCoreBundle/EventListener/BlockCacheResponseListener.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    public function configureBlockCache(FilterResponseEvent $event)
+    {
+        $view = $event->getRequest()->attributes->get('view');
+        if (!$view instanceof CachableView || !$view instanceof BlockView) {
+            return;
+        }
+
+        if (!$this->enableViewCache || !$view->isCacheEnabled()) {
```
Comment (lolautruche) [inline eZ/Bundle/EzPublishCoreBundle/EventListener/BlockCacheResponseListener.php]: Should be a `&&` to stick with global view cache setting (like https://github.com/ezsystems/ezpublish-kernel/pull/1472)

Comment (lolautruche) [inline eZ/Bundle/EzPublishCoreBundle/EventListener/BlockCacheResponseListener.php]: OK forget it, code is right :-)

**Comment by lolautruche** [comment]
+1

File: eZ/Bundle/EzPublishCoreBundle/EventListener/BlockCacheResponseListener.php (hunk 2 above)
Comment (bdunogier) [inline eZ/Bundle/EzPublishCoreBundle/EventListener/BlockCacheResponseListener.php]: Invalid remark as discussed, right @lolautruche ?

Comment (emodric) [inline eZ/Bundle/EzPublishCoreBundle/EventListener/BlockCacheResponseListener.php]: It is :)

The event handler should quit if global view cache is NOT enabled OR if cache in `$view` is NOT enabled.

Comment (lolautruche) [inline eZ/Bundle/EzPublishCoreBundle/EventListener/BlockCacheResponseListener.php]: Yes, invalid remark.

**Comment by emodric** [comment]
\o/

Thanks @bdunogier @lolautruche!

**Pull request merged**